### PR TITLE
ovl_Obj_Yasi

### PR DESCRIPTION
**Pull request by MrPolymorph**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_Obj_Yasi/z_obj_yasi.c [hunk 1]
```diff
@@ -28,25 +30,69 @@ const ActorInit Obj_Yasi_InitVars = {
...
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 400, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 800, ICHAIN_STOP),
 };
 
-#endif
+void ObjYasi_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjYasi* this = THIS;
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Yasi/z_obj_yasi.c:40]: We like to leave an empty line between the variables declarations and the actual code
```suggestion
    ObjYasi* this = THIS;

    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
```

File: src/overlays/actors/ovl_Obj_Yasi/z_obj_yasi.c [hunk 2]
```diff
@@ -28,25 +28,69 @@ const ActorInit Obj_Yasi_InitVars = {
...
+    if ((this->dyna.actor.params & 1) != 0) {
+        this->dyna.actor.scale.x = 0.2f;
+        this->dyna.actor.scale.z = 0.2f;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Yasi/ObjYasi_Init.s")
+void ObjYasi_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjYasi* this = THIS;
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Yasi/z_obj_yasi.c:54]:
```suggestion
    ObjYasi* this = THIS;

    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
```

File: src/overlays/actors/ovl_Obj_Yasi/z_obj_yasi.c [hunk 3]
```diff
@@ -28,25 +28,69 @@ const ActorInit Obj_Yasi_InitVars = {
...
+            }
+        }
+        this->dyna.actor.home.rot.y = GET_PLAYER(globalCtx)->actor.shape.rot.y;
+        this->dyna.actor.home.rot.x = 400;
+        this->dyna.actor.home.rot.z = 0;
+    }
+    temp = (s16)(this->dyna.actor.shape.rot.x * 0.1f);
+    this->dyna.actor.home.rot.x -= temp;
+    this->dyna.actor.shape.rot.x = this->dyna.actor.shape.rot.x +
+                                   ((s16)(this->dyna.actor.home.rot.x - ((s16)(this->dyna.actor.shape.rot.x * 0.08f))));
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Yasi/z_obj_yasi.c:78]: Would using `+=` here match?
Also, the other operand looks like the `BINANG_SUB` macro
```suggestion
    this->dyna.actor.shape.rot.x += BINANG_SUB(this->dyna.actor.home.rot.x, (s16)(this->dyna.actor.shape.rot.x * 0.08f));
```

File: src/overlays/actors/ovl_Obj_Yasi/z_obj_yasi.c [hunk 4]
```diff
@@ -28,25 +28,69 @@ const ActorInit Obj_Yasi_InitVars = {
...
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Yasi/ObjYasi_Draw.s")
+void ObjYasi_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    ObjYasi* this = THIS;
+    Matrix_InsertTranslation(this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.y, this->dyna.actor.world.pos.z,
+                             0);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Yasi/z_obj_yasi.c:84]: Many `Matrix_` functions takes either a `MTXMODE_NEW` or `MTXMODE_APPLY` as the last argument

```suggestion
    ObjYasi* this = THIS;

    Matrix_InsertTranslation(this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.y, this->dyna.actor.world.pos.z,
                             MTXMODE_NEW);
```

File: src/overlays/actors/ovl_Obj_Yasi/z_obj_yasi.c [hunk 5]
```diff
@@ -28,25 +28,69 @@ const ActorInit Obj_Yasi_InitVars = {
...
+
+    if (this->dyna.actor.shape.rot.x != 0) {
+        Matrix_RotateY(this->dyna.actor.home.rot.y, 1);
+        Matrix_InsertXRotation_s(this->dyna.actor.shape.rot.x, 1);
+        Matrix_RotateY((s16)(this->dyna.actor.shape.rot.y - this->dyna.actor.home.rot.y), 1);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Yasi/z_obj_yasi.c:89]:
```suggestion
        Matrix_RotateY(this->dyna.actor.home.rot.y, MTXMODE_APPLY);
        Matrix_InsertXRotation_s(this->dyna.actor.shape.rot.x, MTXMODE_APPLY);
        Matrix_RotateY((s16)(this->dyna.actor.shape.rot.y - this->dyna.actor.home.rot.y), MTXMODE_APPLY);
```

File: src/overlays/actors/ovl_Obj_Yasi/z_obj_yasi.c [hunk 6]
```diff
@@ -28,25 +28,69 @@ const ActorInit Obj_Yasi_InitVars = {
...
+    } else {
+        Matrix_RotateY(this->dyna.actor.shape.rot.y, 1);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Yasi/z_obj_yasi.c:91]:
```suggestion
        Matrix_RotateY(this->dyna.actor.shape.rot.y, MTXMODE_APPLY);
```

File: src/overlays/actors/ovl_Obj_Yasi/z_obj_yasi.c [hunk 7]
```diff
@@ -28,25 +28,69 @@ const ActorInit Obj_Yasi_InitVars = {
...
+    }
+
+    Matrix_Scale(0.1f, 0.1f, 0.1f, 1);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Yasi/z_obj_yasi.c:94]:
```suggestion
    Matrix_Scale(0.1f, 0.1f, 0.1f, MTXMODE_APPLY);
```

**Review by AngheloAlf — COMMENTED** [review]
Good job on your first file.
I left a few comments about code styling

File: src/overlays/actors/ovl_Obj_Yasi/z_obj_yasi.c (hunk 3 above)
Comment (AngheloAlf) [reply src/overlays/actors/ovl_Obj_Yasi/z_obj_yasi.c:78]: You forgot to add the macro to the decompme 
https://decomp.me/scratch/644QW

**Comment by MrPolymorph** [comment]
> Good job on your first file. I left a few comments about code styling

Thanks :)

File: src/overlays/actors/ovl_Obj_Yasi/z_obj_yasi.c (hunk 3 above)
Comment (MrPolymorph) [reply src/overlays/actors/ovl_Obj_Yasi/z_obj_yasi.c]: of course. Thanks :) looks good now!

File: src/overlays/actors/ovl_Obj_Yasi/z_obj_yasi.c [hunk 8]
```diff
@@ -28,25 +28,72 @@ const ActorInit Obj_Yasi_InitVars = {
...
+                sp1C.x = this->dyna.actor.world.pos.x;
+                sp1C.y = this->dyna.actor.world.pos.y + 280.0f;
+                sp1C.z = this->dyna.actor.world.pos.z;
+                Item_DropCollectible(globalCtx, &sp1C, ITEM00_NUTS_1);
+            }
+        }
+        this->dyna.actor.home.rot.y = GET_PLAYER(globalCtx)->actor.shape.rot.y;
+        this->dyna.actor.home.rot.x = 400;
+        this->dyna.actor.home.rot.z = 0;
+    }
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Obj_Yasi/z_obj_yasi.c:76]: This block appears to be to do with when you get an item drop. I looked into this, and it turns out that Player will set the home.rot.z to 1 if you roll close enough to the tree, which then triggers both the decaying swaying motion and the item drop chance. As such, I think we can tread home.rot.z as a bool, and replace `this->dyna.actor.params < 0` by a define like `CAN_DROP_ITEM(thisx)` (or `CAN_DROP_NUT(thisx)` if you prefer to be explicit).

**Review by EllipticEllipsis — COMMENTED** [review]
Good job on this, a couple of suggestions to polish off the documentation

**Comment by EllipticEllipsis** [comment]
This needs a format.

**Review by AngheloAlf — APPROVED** [review]
Remember to run the formatter

File: src/overlays/actors/ovl_Obj_Yasi/z_obj_yasi.c [hunk 9]
```diff
@@ -5,17 +5,19 @@
  */
 
 #include "z_obj_yasi.h"
+#include "objects/object_obj_yasi/object_obj_yasi.h"
 
 #define FLAGS 0x00000000
 
 #define THIS ((ObjYasi*)thisx)
 
+#define CAN_DROP_NUT(thisx) (thisx->params < 0)
```
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Yasi/z_obj_yasi.c:14]:
```suggestion
#define CAN_DROP_NUT(thisx) ((thisx)->params < 0)
```

**Review by hensldm — APPROVED** [review]
Nice work on this actor. Just one minor suggestion.

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**